konan_sdk/endpoints/konan_endpoints.py

```diff
@@ -0,0 +1,44 @@
+from konan_sdk.endpoints.base_endpoint import KonanBaseEndpoint
+
+
+class LoginEndpoint(KonanBaseEndpoint):
+    name = 'login'
+
+    def __init__(self, api_url, user) -> None:
+        super().__init__(api_url=api_url, user=user)
+
+    @property
+    def endpoint_path(self):
+        return '/api/auth/login/'
+
+
+class RefreshTokenEndpoint(KonanBaseEndpoint):
+    name = 'refresh_token'
+
+    def __init__(self, api_url, user) -> None:
+        super().__init__(api_url=api_url, user=user)
+
+    @property
+    def endpoint_path(self):
+        return '/api/auth/token/refresh/'
+
+
+class PredictionEndpoint(KonanBaseEndpoint):
+    name = 'predict'
+
+    def __init__(self, deployment_uuid, api_url, user) -> None:
```

Comment:
Contrary to my earlier comment, actually best practice is to have child classes _not_ change a parent class' method's signature.

To work around child `*Endpoint` classes requiring different `__init__()` arguments, you can use the `*args` and `**kwargs` in `BaseEndpoint.__init__()` to signify that child classes may add their extra arguments (`*args`) and key-word arguments (`**kwargs`). Then, you may add any extra arguments or keyword arguments in child `*Endpoint.__init__()` methods.

If you want to ensure maximum compatibitly in code using these classes, however, it is better to _not_ add any extra "arguments" in child classes' `__init__()` methods, and instead use "keyword arguments" with sane defaults. You may also add extra checks in `*Endpoint.__init__()` to ensure that the instanitating code indeed passed a valid parameter to the key-word argument and that it is not still defaulted. 

This will allow us to write a code like this later on:

```
    EndpointClasses = [LoginEndpoint, RefreshTokenEndpoint, PredictionEndpoint]
    endpoint_objects = [EC(api_url, user, deployment_uuid="<deployment-uuid>") for EC in EndpointClasses]
```

Check this [link](https://stackoverflow.com/questions/6034662/python-method-overriding-does-signature-matter) for some additional info.

```suggestion
    def __init__(self, api_url, user, deployment_uuid=None, *args, **kwargs) -> None:
        if deployment_uuid is None:
            raise ValueError("A valid deployment_uuid must be specified")
```